src/mpi/simulation.rs

```diff
@@ -80,6 +100,10 @@ impl Simulation {
                     agent.curr_act().act_type.clone(),
                 ),
             );
+
+            Self::find_route(router, agent);
```

Comment:
See comment in `find_route` function. 

If this method returns `(route, tavel_time)` this could then be passed to `agent.add_leg(route, travel_time)` or something like this.